mitodl/micromasters
# Refactored Education frontend components

#### What are the relevant tickets?

Part of #1498
#### What's this PR do?

Removes `EducationDialog` and `EducationDisplay` and integrates that functionality into `EducationForm`. It's meant to make `EducationForm` more similar to `EmploymentForm` to make future refactoring easier.
#### How should this be manually tested?

Go through the profile editing steps, including `/profile` and `/learner/username`, including editing, adding and deleting education and work history. Nothing should have changed compared to master

However I did change the month/year formatting for course states, which is now  MM/YYYY to make it consistent with the profile. Any comments here?
